base/file.jl

```diff
@@ -637,14 +637,21 @@ function chmod(path::AbstractString, mode::Integer; recursive::Bool=false)
     nothing
 end
 
+sentinel_code(i::Integer) = i
+sentinel_code(n::Nothing) = -1
+sentinel_code(n::Missing) = -2
+sentinel_code(a) = ArgumentError("$a is not an integer, nothing, or missing")
+
 """
-    chown(path::AbstractString, owner::Integer, group::Integer=-1)
+    chown(path::AbstractString; owner = nothing, group = nothing)
 
-Change the owner and/or group of `path` to `owner` and/or `group`. If the value entered for `owner` or `group`
-is `-1` the corresponding ID will not change. Only integer `owner`s and `group`s are currently supported.
+Change the owner and/or group of `path` to `owner` and/or `group`. If left as
+`nothing`, then the they won't be changed. If set to `missing`, they will be
```

Comment:
How about using something like `owner = :remove` instead of `missing`? We could also use e.g. `:preserve` instead of `nothing`, that would be more explicit as `nothing` could also mean "remove".

Reply:
Sure that sounds good.